src/api/admin.ts

```diff
@@ -0,0 +1,61 @@
+import firebase from "firebase/app";
+import { UserInfo } from "firebase";
+import _ from "lodash";
+
+export interface VenueInput {
+  name: string;
+  bannerImageFile: FileList;
+  logoImageFile: FileList;
+  tagline: string;
+  longDescription: string;
+}
+
+type FirestoreVenueInput = Omit<
+  VenueInput,
+  "bannerImageFile" | "logoImageFile"
+> & {
+  bannerImageUrl: string;
+  logoImageUrl: string;
+};
+
+export const createUrlSafeName = (name: string) => name.replace(/\W/g, "");
```

Comment:
Should we send this to the backend instead of recomputing it?

Reply:
don't think so. I think backend should compute document id's (though doesn't make much of a difference in this case).